### PR TITLE
Add missing [[maybe_unused]] annotations in unit tests

### PR DESCRIPTION
**Pull request by lpbeliveau-silabs**
Added [[maybe_unused]] to allow building unit test and Silabs apps without logs

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/platform/silabs/SoftwareFaultReports.cpp [hunk 1]
```diff
@@ -89,18 +89,18 @@ void OnSoftwareFaultEventHandler(const char * faultRecordString)
...
-    uint32_t bfar  = SCB->BFAR;
-    uint32_t r0    = sp[0];
-    uint32_t r1    = sp[1];
-    uint32_t r2    = sp[2];
-    uint32_t r3    = sp[3];
-    uint32_t r12   = sp[4];
-    uint32_t lr    = sp[5];
-    uint32_t pc    = sp[6];
-    uint32_t psr   = sp[7];
+    [[maybe_unused]] uint32_t cfsr  = SCB->CFSR;
```
Comment (cecille) [inline examples/platform/silabs/SoftwareFaultReports.cpp:92]: Given that this is guarded by an ifdef, why would these be unused? Is there a case where SILABS_LOG_ENABLED is on, but the chip log functions are noops?

Comment (lpbeliveau-silabs) [reply examples/platform/silabs/SoftwareFaultReports.cpp:92]: if we disable only error logging

Comment (lpbeliveau-silabs) [reply examples/platform/silabs/SoftwareFaultReports.cpp:92]: Added a check on chip_logging in .gni files to ensure turning all logging off removes dead code

Comment (cecille) [reply examples/platform/silabs/SoftwareFaultReports.cpp:92]: OH. gotcha.

**Review by cecille — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**